tests/test_machine.py

```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+
+import pytest
+
+import mkosi.machine as machine
+
+
+@pytest.mark.parametrize("verb", ["qemu", "boot"])
+@pytest.mark.skipif(os.getuid() != 0, reason="Must be invoked as root.")
+@pytest.mark.integration
+def test_run(verb: str) -> None:
+    boot = ["-d", "fedora", "-f", "--autologin"]
+
+    if verb == "qemu":
+        boot += ["--bootable", "--qemu-headless", "--hostonly-initrd", "--network-veth", "--ssh", verb]
+    if verb == ("boot"):
+        boot += [verb]
+
+    with machine.Machine(boot) as m:
+        p = m.run(["echo", "This is a test."])
```

Comment:
There's room for a lot more testing here:

- What happens if we run a command that fails?
- What happens if we run the same command a bunch of times in a loop?
- What if we run a command that never exits?
- What happens if we try to run a command before the machine has booted or after it has shut down?